### DISCUSSION TITLE
Non-Euclidean possibilities

### DISCUSSION THREAD

**Discussion by MasterLaplace** [MasterLaplace/Engine-3D#3] [Ideas]
Non-Euclidean geometry is a type of geometry where the basic axioms and assumptions of Euclidean geometry do not hold. In a game engine, this can be used to create unique and interesting gameplay experiences, such as impossible structures, warped space, and optical illusions.

There are several ways to implement non-Euclidean geometry in a game engine, including:

1. Portal rendering: This technique uses portals to link different areas of a game world together, allowing players to move between them seamlessly. The portals can be placed in such a way that they warp the space around them, creating impossible structures and optical illusions.

2. Non-Euclidean level design: In this approach, the game world is designed to intentionally violate the laws of Euclidean geometry. This can include non-orthogonal angles, non-parallel lines, and other geometric distortions that create a surreal and otherworldly environment.

3. Hyperbolic geometry: Hyperbolic geometry is a type of non-Euclidean geometry where the parallel postulate is violated. In a game engine, hyperbolic geometry can be used to create environments with infinite space, where players can move in a straight line and eventually return to their starting point.

4. Fractal geometry: Fractal geometry is a type of non-Euclidean geometry that describes complex, self-similar patterns that repeat at different scales. In a game engine, fractal geometry can be used to create natural-looking terrain and structures that are infinitely detailed.

Overall, incorporating non-Euclidean geometry into a game engine can lead to unique and fascinating gameplay experiences that push the boundaries of what players expect from a video game.

**Comment by M7T5M3P** [comment]
Thanks for sharing this information! Implementing non-Euclidean geometry in game engines can indeed lead to innovative and exciting gameplay experiences. I appreciate the different ways you've mentioned, including portal rendering, non-Euclidean level design, hyperbolic geometry, and fractal geometry. This knowledge will certainly be useful for game developers looking to create new and unique game worlds.
